pkg/operator/options/options.go

```diff
@@ -0,0 +1,134 @@
+	fs.DurationVar(&o.AssumeRoleDuration, "assume-role-duration", env.WithDefaultDuration("ASSUME_ROLE_DURATION", 15*time.Minute), "Duration of assumed credentials in minutes. Default value is 15 minutes. Not used unless aws.assumeRole set.")
+	fs.StringVar(&o.ClusterCABundle, "cluster-ca-bundle", env.WithDefaultString("CLUSTER_CA_BUNDLE", ""), "Cluster CA bundle for nodes to use for TLS connections with the API server. If not set, this is taken from the controller's TLS configuration.")
+	fs.StringVar(&o.ClusterName, "cluster-name", env.WithDefaultString("CLUSTER_NAME", ""), "[REQUIRED] The kubernetes cluster name for resource discovery.")
+	fs.StringVar(&o.ClusterEndpoint, "cluster-endpoint", env.WithDefaultString("CLUSTER_ENDPOINT", ""), "The external kubernetes cluster endpoint for new nodes to connect with. If not specified, will discover the cluster endpoint using DescribeCluster API.")
+	fs.StringVar(&o.isolatedVPC, "isolated-vpc", env.WithDefaultString("ISOLATED_VPC", "false"), "If true, then assume we can't reach AWS services which don't have a VPC endpoint. This also has the effect of disabling look-ups to the AWS pricing endpoint.")
+	fs.Float64Var(&o.VMMemoryOverheadPercent, "vm-memory-overhead-percent", env.WithDefaultFloat64("VM_MEMORY_OVERHEAD_PERCENT", 0.075), "The VM memory overhead as a percent that will be subtracted from the total memory for all instance types.")
+	fs.StringVar(&o.InterruptionQueueName, "interruption-queue-name", env.WithDefaultString("INTERRUPTION_QUEUE_NAME", ""), "Interruption queue is disabled if not specified. Enabling interruption handling may require additional permissions on the controller service account. Additional permissions are outlined in the docs.")
+	fs.IntVar(&o.ReservedENIs, "reserved-enis", env.WithDefaultInt("RESERVED_ENIS", 0), "Reserved ENIs are not included in the calculations for max-pods or kube-reserved. This is most often used in the VPC CNI custom networking setup https://docs.aws.amazon.com/eks/latest/userguide/cni-custom-network.html.")
+}
+
+func (o *Options) Parse(fs *flag.FlagSet, args ...string) error {
+	if err := fs.Parse(args); err != nil {
+		if errors.Is(err, flag.ErrHelp) {
+			os.Exit(0)
+		}
+		return fmt.Errorf("parsing flags, %w", err)
+	}
+
+	// Check if each option has been set. This is a little brute force and better options might exist,
+	// but this only needs to be here for one version
+	o.setFlags = map[string]bool{}
+	cliFlags := sets.New[string]()
+	fs.Visit(func(f *flag.Flag) {
+		cliFlags.Insert(f.Name)
+	})
+	fs.VisitAll(func(f *flag.Flag) {
+		envName := strings.ReplaceAll(strings.ToUpper(f.Name), "-", "_")
+		_, ok := os.LookupEnv(envName)
+		o.setFlags[f.Name] = ok || cliFlags.Has(f.Name)
+	})
+
+	if err := o.Validate(); err != nil {
+		return fmt.Errorf("validating options, %w", err)
+	}
+
+	// Parse options which can't be directly parsed by flag
+	o.IsolatedVPC = (o.isolatedVPC == "true")
```

Comment:
Can you explain this? Or link to a file or description of this behavior? This isn't going to be immediately evident to anyone reading it, so it should have some link to documentation if this is needed for the behavior.

Reply:
Do you mean why this can't be parsed via BoolVar and it has to be parsed as a string and then converted? The go docs didn't really have a good description of it, I can see if I can find one but worst case I found some examples.

Reply:
I think I got caught up by `BoolVar` being the only one of the `TypeVar` functions on `flag` that doesn't take an argument (without calling it out), but I guess `--isolated-vpc` probably shouldn't take an argument (i.e. `--isolated-vpc` instead of `--isolated-vpc true`.

Reply:
A counterpoint to this would be that there is no longer an explicit way to set `isolated-vpc` to false via the CLI flags. This isn't an issue if we consider environment variables and cli flags as mutually exclusive (a user will only specify one and not the other), but otherwise it makes it impossible to explicitly set `isolated-vpc` to false when `ISOLATED_VPC=true` exists as an env var. I'm honestly not sure if this is contrived / a non-issue or if it's worth considering.

Reply:
After sleeping on it I think the correct solution is to use a `StringVar` for `IsolatedVPC` so we can explicitly set it to true or false via the CLI flags. I think the consistent merge semantics with environment variables outweigh breaking consistency with how bools are normally handled with flags.

Reply:
This is being addressed by a wrapper to `FlagSet` implemented in karpenter-core.